### ISSUE TITLE
NPMI scorer does not take into account `min_count`

### ISSUE THREAD

**Issue by kikoaumond** [RaRe-Technologies/gensim#2086]
in phrases.py, npmi_scorer does not take into account min_count when scoring bigrams.

I suggest the following alternative for taking min_count into account when using NPMI scoring.  This function has yielded good results for me; however,  I have not explored other ways to  use min_count:

```
# Custom scoring function to take into account min_counts when using npmi
def npmi_scorer_with_min_count(worda_count, wordb_count, bigram_count, len_vocab, min_count, corpus_word_count):

    if bigram_count < min_count:
        return -1

    pa = worda_count / corpus_word_count
    pb = wordb_count / corpus_word_count
    pab = bigram_count / corpus_word_count
    return log(pab / (pa * pb)) / -log(pab)

```

**Comment by EstevaoUyra** [comment]
The default score also does not takes into account properly. Your function sadly does not correct the problem. The behavior is very strange, I think the min_count should be implemented in the analyze_sentence instead of appearing in the score

**Comment by rafabr4** [comment]
In my tests, the default scorer does take into account the min_count parameter properly. I verified that by changing the parameter many times and looking at the length of the Phraser.phrasegrams dictionary. However, when using 'npmi' scorer, the length never changes by modifying the min_count, only with the threshold parameter.

**Comment by lopusz** [comment]
A while ago, I also noticed that problem with NPMI and sumbmitted a pull request on that
https://github.com/RaRe-Technologies/gensim/pull/2072

**Comment by menshikh-iv** [comment]
Fixed in #2072
